quinn-proto/src/endpoint.rs

```diff
@@ -653,6 +653,18 @@ impl Endpoint {
         }
     }
 
+    /// Reject new incoming connections without affecting existing connections
+    ///
+    /// Convenience short-hand for using
+    /// [`set_server_config`](Self::set_server_config) to update
+    /// [`concurrent_connections`](ServerConfig::concurrent_connections) to
+    /// zero.
+    pub fn reject_new_connections(&mut self) {
+        if let Some(config) = self.server_config.as_mut() {
+            Arc::make_mut(config).concurrent_connections(0);
```

Comment:
Should this me `Arc::make_mut` and also handle a case where the value is used concurrently more explicitly? I don't think clone-on-write is really what's meant here...

Reply:
For the purposes of this code, it seems sufficient to me that `self.server_config` holds the modified config. Seems to me that we don't really care that `Connection`s are holding on to a "stale" `ServerConfig` that still allows more `concurrent_connections`.

Reply:
I misread what `Arc::make_mut` actually does. Yeah, this design is good!